pkg/volume/util.go

```diff
@@ -398,3 +398,44 @@ func GetPVCMatchLabel(pvc *v1.PersistentVolumeClaim, key string) (string, error)
 	}
 	return "", fmt.Errorf("Persistent volume claim (%v): %v key not found in the matchLabels part", pvc.Name, key)
 }
+
+// GetPVCMatchExpression returns:
+// - either (setOfValues, nil) for all matching (key, operator) from the matchExpressions Selector part of the PVC
+// - or (emptySet, error) in case the operator or the key is missing in the matchExpressions Selector part of the PVC
+// Example:
+// selector:
+//     matchExpressions:
+//       - key: failure-domain.beta.kubernetes.io/zone
+//         operator: In
+//         values:
+//           - us-east-1a
+//           - us-east-2a
+//           - us-east-3a
+//       - key: failure-domain.beta.kubernetes.io/zone
+//             operator: In
+//             values:
+//               - us-east-3a
+//               - us-east-4a
+// Returns (sets.String{"us-east-1a": sets.Empty{}, "us-east-2a": sets.Empty{}, "us-east-3a": sets.Empty{}, "us-east-4a": sets.Empty{}}, nil)
```

Comment:
this example is wrong, docs of `MatchExpressions` says that "The requirements are ANDed", therefore it must match both expressions, i.e. only "us-east-3a" should be returned.

And beware of `NotIn` operator, it behaves differently than `In` here.